protocol/nats_jetstream/v2/sender.go

```diff
@@ -44,21 +44,28 @@ func NewSender(url, stream, subject string, natsOpts []nats.Option, jsmOpts []na
 
 // NewSenderFromConn creates a new protocol.Sender which leaves responsibility for opening and closing the NATS
 // connection to the caller
+// The stream parameter is only needed if auto-creation of the stream is needed when the stream does not exist.
 func NewSenderFromConn(conn *nats.Conn, stream, subject string, jsmOpts []nats.JSOpt, opts ...SenderOption) (*Sender, error) {
 	jsm, err := conn.JetStream(jsmOpts...)
 	if err != nil {
 		return nil, err
 	}
 
-	streamInfo, err := jsm.StreamInfo(stream, jsmOpts...)
-
-	if streamInfo == nil || err != nil && err.Error() == "stream not found" {
-		_, err = jsm.AddStream(&nats.StreamConfig{
-			Name:     stream,
-			Subjects: []string{stream + ".*"},
-		})
-		if err != nil {
-			return nil, err
+	// A Stream parameter is not needed for a send operation.
+	// A subject is all that is needed.
+	// Below, a stream is created with default stream config which may not be desired.
+	// It may be that the intention is for the call to fail if a stream does not exist.
+	if stream != "" {
+		streamInfo, err := jsm.StreamInfo(stream, jsmOpts...)
+
+		if streamInfo == nil || err != nil && err.Error() == "stream not found" {
```

Comment:
I know you're reusing the existing implementation here, but couple questions:

1) Could there be a scenario where both, streamInfo and err are nil? If not, then just checking the error might be enough here
2) Instead of asserting on err.Error() please use typed error checking `ErrStreamNotFound`

Reply:
Looking at the source, it looks like there is only one line that returns an info object.  All other lines return a non-nil error.  Once again, I only added an if statement around the existing code.  I can make that change if you like.  Also, if I were to embark on a new implementation based on the new jetstream api, I might simply make changes there.

Reply:
Yeah, probably better to fix in the other PR when using the new JetStream API. I was just bringing those concerns up here because I wasn't part of the initial reviews and by asking these questions we have opportunities to simplify the code base/not complicate it further for our future selves :)

Appreciate your work!